core/src/main/scala/org/apache/spark/partial/SumEvaluator.scala

```diff
@@ -56,9 +56,12 @@ private[spark] class SumEvaluator(totalOutputs: Int, confidence: Double)
       val confFactor = {
         if (counter.count > 100) {
           new NormalDistribution().inverseCumulativeProbability(1 - (1 - confidence) / 2)
-        } else {
+        } else if (counter.count > 1) {
           val degreesOfFreedom = (counter.count - 1).toInt
           new TDistribution(degreesOfFreedom).inverseCumulativeProbability(1 - (1 - confidence) / 2)
+        } else {
+          // No way to meaningfully estimate confidence, so we signal no particular confidence interval
+          Double.PositiveInfinity
```

Comment:
But the relevant computation is:
```
scala> Double.NaN * Double.PositiveInfinity
res1: Double = NaN
```

The interval should not be `NaN`.

If count is 1 then the sample variance is NaN along with everything else that's a function of a second moment.